### PR TITLE
Remove direct use of Object Manager

### PR DESCRIPTION
**Pull request by AnshuMishra17**
Description
Remove direct use of Object Manager by adding dependency in constructor.

Manual testing scenarios
Go to Content -> CMS Pages
Delete CMS pages from edit page and massaction as well.

### CODE REVIEW

File: app/code/Magento/Cms/Controller/Adminhtml/Page/Delete.php [hunk 1]
```diff
@@ -30,8 +49,7 @@ public function execute()
             $title = "";
             try {
                 // init model and delete
-                $model = $this->_objectManager->create(\Magento\Cms\Model\Page::class);
-                $model->load($id);
+                $model = $this->cmsPage->load($id);
```
Comment (ihor-sviziev) [inline app/code/Magento/Cms/Controller/Adminhtml/Page/Delete.php]: 1. I think in this case it's better to use repository to retrieve and then delete this cms page
2. Loading model is not correct way. It's better to use \Magento\Cms\Model\PageFactory and create one instance of model your your needs (if really needed)

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: app/code/Magento/Cms/Controller/Adminhtml/Page/Delete.php (hunk 1 above)
Comment (AnshuMishra17) [reply app/code/Magento/Cms/Controller/Adminhtml/Page/Delete.php]: Done

**Comment by AnshuMishra17** [comment]
I have made the changes @ihor-sviziev can you please review?

File: app/code/Magento/Cms/Controller/Adminhtml/Page/Delete.php [hunk 2]
```diff
@@ -14,6 +16,23 @@ class Delete extends \Magento\Backend\App\Action
...
+    /**
+     * @param Context $context
+     * @param \Magento\Cms\Api\PageRepositoryInterface $cmsPage
+     */
+    public function __construct(
+        Context $context,
+        \Magento\Cms\Api\PageRepositoryInterface $cmsPage
+    ) {
+        parent::__construct($context);
+        $this->cmsPage = $cmsPage ?: \Magento\Framework\App\ObjectManager::getInstance()->get(\Magento\Cms\Api\PageRepositoryInterface::class);
```
Comment (ihor-sviziev) [inline app/code/Magento/Cms/Controller/Adminhtml/Page/Delete.php]: It's not "cms page", now it's "cms page repository", please rename it

File: app/code/Magento/Cms/Controller/Adminhtml/Page/Delete.php [hunk 3]
```diff
@@ -14,6 +16,23 @@ class Delete extends \Magento\Backend\App\Action
...
+     */
+    private $cmsPage;
+    
+    /**
+     * @param Context $context
+     * @param \Magento\Cms\Api\PageRepositoryInterface $cmsPage
+     */
+    public function __construct(
+        Context $context,
+        \Magento\Cms\Api\PageRepositoryInterface $cmsPage
```
Comment (ihor-sviziev) [inline app/code/Magento/Cms/Controller/Adminhtml/Page/Delete.php]: Please make new dependency optional (you already added fallback to object manager) due to backward compatibility reasons

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

**Comment by ihor-sviziev** [comment]
@AnshuMishra17
1. please fix these issues and combine all your changes into one commit
2. unit tests are failing right now, please check them

File: app/code/Magento/Cms/Controller/Adminhtml/Page/Delete.php (hunk 2 above)
Comment (AnshuMishra17) [reply app/code/Magento/Cms/Controller/Adminhtml/Page/Delete.php]: Done

File: app/code/Magento/Cms/Controller/Adminhtml/Page/Delete.php (hunk 3 above)
Comment (AnshuMishra17) [reply app/code/Magento/Cms/Controller/Adminhtml/Page/Delete.php]: Done.

**Comment by AnshuMishra17** [comment]
@ihor-sviziev I have made the necessary modifications. It would be great if you can verify and let me know if there are still any issues.
I am new here, I will be thankful to you as well if you can guide me towards the correct approach and steps.
Also, how can I check what are the issues with unit tests?

**Comment by AnshuMishra17** [comment]
@ihor-sviziev How can track the issue here?
I have checked one report and it is saying "Error: Call to a member function getTitle() on null".
But I have checked using my local setup manual testing by printing the title and the deleted page title is getting print.
How can I make sure in which scenario the test is getting failed?

**Comment by Jakhotiya** [comment]
Hi @AnshuMishra17 you will need to modify this unit test "\Magento\Cms\Test\Unit\Controller\Adminhtml\Page\DeleteTest". You have refactored the Main Controller class but you will also need to cover all those changes and update this test class.
For example you will add pageRepository mock to your test class and use that.

**Comment by AnshuMishra17** [comment]
@ihor-sviziev @Jakhotiya How to modify Magento\Cms\Test\Unit\Controller\Adminhtml\Page\DeleteTest ? If anyone of you can guide me that will be really helpful to me.

**Comment by Jakhotiya** [comment]
@AnshuMishra17 have you run this unit test on your local? Seems like tests are failing on CI. Can you run all unit tests on local? If you are not familiar with how to do that, [unit testing in magento 2](https://devdocs.magento.com/guides/v2.2/test/unit/unit_test_execution_cli.html) should help.
Sorry for the delay in response.

**Pull request closed without merging**